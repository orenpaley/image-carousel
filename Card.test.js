
import { render } from "@testing-library/react";
import Card from "./Card";
import image1 from "./image1.jpg";

describe("Smokeshots and Snapshots", () => {
  const prop = {msg: "test test test", caption: image1}
  it("renders without crashing", () => {
    render(<Card props={prop}/>);
  });
  it("matches snapshot", function() {
    const {asFragment} = render(<Card props={prop} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("matches snapshot", function() {
    const {asFragment} = render(<Card msg="New Test" caption={image1} />);
    expect(asFragment()).toMatchSnapshot();
  });
})

