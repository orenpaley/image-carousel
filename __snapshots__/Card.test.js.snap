// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smokeshots and Snapshots matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    />
    <img
      class="Card-image"
    />
    <small
      class="Card-small"
    >
      Image  of .
    </small>
  </div>
</DocumentFragment>
`;

exports[`Smokeshots and Snapshots matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      image1.jpg
    </h4>
    <img
      alt="image1.jpg"
      class="Card-image"
    />
    <small
      class="Card-small"
    >
      Image  of .
    </small>
  </div>
</DocumentFragment>
`;

exports[`Smokeshots and Snapshots matches snapshot 3`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      bad
    </h4>
    <img
      alt="bad"
      class="Card-image"
    />
    <small
      class="Card-small"
    >
      Image  of .
    </small>
  </div>
</DocumentFragment>
`;
